approach_data_structure_task.md

```diff
@@ -0,0 +1,168 @@
+
+In this scenerio, we can use the general word for the people who use the system - `user`. In the previous scenerio we called the `user` entity by the more specific name `author`but essentially they are the same.
+
+We know that the `user` has a relationship with a `topic` so we draw a line between them. 
+
+Question 1: asked in both directions
+Can a `user` register for more than one `topic`?
+Can a `topic` be chosen by more than one `user`?
+
+The answer 'yes' to this question (in both directions) is draws like this:
+![](6.jpg)
+
+
+Question 2 (The Million Question): 
+Can and should one single `topic` refer to a million `user`?
+Can and should one single `user` refer to a million `topic`?
+
+Hmm, seems we have a real problem here! A single `user` cannot reference million `topic`, and vice versa. Is the universe going to collapse into itself?! 
+
+Worry not, because we have an elegant solution for cases like this. We'll simply create a new entity! We will place it in between the other two entities to solve the out-of-control referencing of a million in both directions. We call this "revealing the entity", as sometimes an entity will be hidden in the requirements and not get an explicit name. In our case, giving it a name shouldn't be too hard. We can call it `membership` and it will represent a specific registration (or a membership) of a user to a topic. Now every `user` has only one single `membership` per `topic`.
+
+![](7.jpg)
+
+By having a special entity for capturing the membership, we can actually add more meta-data like the state of the membership (is the user an active member, pending or even blocked)or timestamp (when the `membership` was created) and so on.
+
+Side note: It's worth mentioning that this `membership` entity and its references is the base concept for the Organic Groups module in Drupal.
+But we must continue, it seems that our client has more needs for their premium website.
+
+> Registration should expire after a year. The member will get a reminder emails after 3 month, 1 month and 1 day before expiration. If not renewed, the membership should be set to pending.
+ 
+What information do we need in order to send the reminder emails?
+
+We need to know, of course, when the membership began, so we can calculate when 9 months has passed to send the first reminder email. In technical words, the beginning date or any other date that represent that something has occured in the system is called a `timestamp`.
+
+But this information is not enough. Think about a situation where the user decides not to renew his registration. What happened to his membership? Does it go away, get deleted from the system? Well, usually we don’t delete content from the system, but we can mark it as inactive.
+
+So in order to send the user a reminder, we need to know two things:  the membership created date (`timestamp`) and if it is active or not (`status`).
+
+![](9.jpg)
```

Comment:
Indeed, shouldn't as circles - because they are a `property` of the `entity`. Can be in a follow up, for when we replace the images to something like we have in http://www.gizra.com/content/og8-development-mindset/